src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -93,68 +94,55 @@ def get_expiration_warning_message(self, loop_current_time):
-        self.sender_email = config['sender']['email']
-        if self.sender_email in self.email_dicts.keys():
-            # The sender is also a recipient.
-            self.email_dicts[self.sender_email]['is_sender'] = True
+            # Mark the gpgmailer configured sender key.
+            if fp_to_key_dict[fingerprint]['fingerprint'] == config['sender']['fingerprint']:
+                self.sender_key = key_dict
 
-            if config['sender']['fingerprint'] not in recipient_fingerprints:
-                raise RecipientEmailCollision('Email %s is already configured with a different key.' % recipient['email'])
+            self.recipient_keys[fp_to_key_dict[fingerprint]['fingerprint']] = key_dict
 
-        else:
-            # The sender is NOT a recipient.
-            email_dict = { 'fingerprint': config['sender']['fingerprint'],
-                'expired_email_sent': False,
-                'expiring_soon_email_sent': False,
-                'is_sender': True,
-                'is_recipient': False}
 
-            self.email_dicts[self.sender_email] = email_dict
-
-
-    # Calculates which recipients keys have expired and which have not. Builds a list of currently
-    #   valid recipients and constructs an expiration warning messages for keys that have expired
+    # Calculates which keys in the keyring have expired and which have not. Builds a list of currently
+    #   valid recipient keys and constructs expiration warning messages for keys that have expired
     #   or will be expiring soon.
     #
     # loop_current_time: The Unix time associated with the main program loop from which all
     #   GPG key expiration checks are based.
     def _calculate_recipient_info(self, loop_current_time):
         self.logger.trace('Recalculating the list of keys that are about to expire.')
-        all_expiration_warning_messages = []
         expired_messages = []
         expiring_soon_messages = []
         valid_recipient_emails = []
         valid_key_fingerprints = []
 
+        # Set the threshold dates for expired/expiring "soon."
         expiration_date = loop_current_time + self.config['key_check_interval'] + \
             self.config['main_loop_duration']
         expiring_soon_date = expiration_date + self.config['expiration_warning_threshold']
 
+        # Calculate all of the expiration information for the sender.
         self.logger.trace('Checking sender key.')
-        sender_expiration_data = self._build_key_expiration_warning_message(self.sender_email,
-            expiration_date, expiring_soon_date)
+        if not self.sender_key:
+            raise NoUsableKeysException("The sender key was not found in the keyring.")
+        else:
+            sender_expiration_data = self._build_key_expiration_warning_message(
+                self.sender_key['fingerprint'], expiration_date, expiring_soon_date)
 
+        # Add expiry messages or raise exception based on sender expiry info generated above.
```

Comment:
A little bit more is done in this 'if' statement. This comment might not be accurate enough. The part that I think is missing is that it gathers e-mails and valid keys.